dsl/jpql/src/main/kotlin/com/linecorp/kotlinjdsl/dsl/jpql/Jpql.kt

```diff
@@ -2518,4 +2518,82 @@ open class Jpql : JpqlDsl {
     fun <T : Any> deleteFrom(entity: Entityable<T>): DeleteQueryWhereStep<T> {
         return DeleteQueryDsl(entity.toEntity())
     }
+
+    /**
+     * Creates an expression that represents the position of the first occurrence of a substring in a string.
+     * If the substring is not found, returns 0. position start with 1.
+     */
+    @SinceJdsl("3.0.0")
+    fun locate(
+        substring: String,
+        string: String,
+        start: Int?,
+    ): Expression<Int> {
+        return locate(Expressions.value(substring), Expressions.value(string), Expressions.value(start))
+    }
+
+    /**
+     * Creates an expression that represents the position of the first occurrence of a substring in a string.
+     * If the substring is not found, returns 0. position start with 1.
+     */
+    @SinceJdsl("3.0.0")
+    fun locate(
+        substring: String,
+        string: Expressionable<String>,
+        start: Int?,
+    ): Expression<Int> {
+        return locate(Expressions.value(substring), string, Expressions.value(start))
+    }
+
+    /**
+     * Creates an expression that represents the position of the first occurrence of a substring in a string.
+     * If the substring is not found, returns 0. position start with 1.
+     */
+    @SinceJdsl("3.0.0")
+    fun locate(
+        substring: Expressionable<String>,
+        string: Expressionable<String>,
+        start: Int?,
+    ): Expression<Int> {
+        return locate(substring.toExpression(), string.toExpression(), Expressions.value(start))
+    }
```

Comment:
It's good to have all the possible cases of a method's parameters because we don't know how the user will use them. But our maintenance costs are too high.

So as a default policy, when creating `DSL` methods, I want to create only two: one where all parameters are `Expressionable` and one where all parameters are of the `default type`.

However, in the case of `locate`, it seems that there are many cases where the user wants to find the substring of a specific column. So for `locate` only, It is okay that `substring: String, string: Expressionable<String>, start: Int?`

Reply:
@shouwn 리뷰 감사합니다.
유지 관리 비용이 많이 든다는 점에 동의합니다.
저도 코드 작성하는 중에 얼마나 사용자 편의를 위해 다양한 메서드를 제공해야할까?에 대해 고민했었는데,
기본 정책을 말씀해주셔서 앞으로는 최대한 반영하겠습니다.